### PR TITLE
feat: add e2e testing for both menu links. Also fixed some e2e stuff.

### PR DESCRIPTION
**Pull request by bradymiller**
feat: add e2e testing for both menu links. Also fixed some e2e stuff.

### CODE REVIEW

File: tests/Tests/E2e/CreateStaffTest.php [hunk 1]
```diff
@@ -31,8 +31,8 @@ protected function tearDown(): void
     public function check_add_user(): void
     {
 
-        echo "SKIPPING THE CHECK ADD USER TEST FOR NOW SINCE BREAKING FOR UNCLEAR REASONS ON PHP 8.3. WILL CONTINUE TO REIMPLENT THIS IN FUTURE\n\n";
-        $this->markTestSkipped('must be revisited.');
+        #echo "SKIPPING THE CHECK ADD USER TEST FOR NOW SINCE BREAKING FOR UNCLEAR REASONS ON PHP 8.3. WILL CONTINUE TO REIMPLENT THIS IN FUTURE\n\n";
+        #$this->markTestSkipped('must be revisited.');
```
Comment (bradymiller) [inline tests/Tests/E2e/CreateStaffTest.php:35]: will remove above commented out lines

**Comment by bradymiller** [comment]
Have added functional testing for all the main menu and user menu links (note skipped the links that require patient/encounter context; these tests can be added in future).

```
Running OpenEMR e2e tests
PHPUnit 10.4.2 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.3.8
Configuration: /var/www/localhost/htdocs/openemr/phpunit.xml

...............................................................  63 / 103 ( 61%)
........................................                        103 / 103 (100%)

Time: 05:56.952, Memory: 38.50 MB

Check Main Menu Links (OpenEMR\Tests\E2e\CheckMainMenuLinks)
 ✔ Check menu link with data set "Calendar menu link"
 ✔ Check menu link with data set "Finder menu link"
 ✔ Check menu link with data set "Flow menu link"
 ✔ Check menu link with data set "Recalls menu link"
 ✔ Check menu link with data set "Messages menu link"
 ✔ Check menu link with data set "Patient -> New/Search menu link"
 ✔ Check menu link with data set "Fees -> Billing Manager menu link"
 ✔ Check menu link with data set "Fees -> Batch Payments menu link"
 ✔ Check menu link with data set "Fees -> Posting Payments menu link"
 ✔ Check menu link with data set "Fees -> EDI History menu link"
 ✔ Check menu link with data set "Modules -> Manage Modules menu link"
 ✔ Check menu link with data set "Modules -> Carecoordination menu link"
 ✔ Check menu link with data set "Procedures -> Procedures Providers menu link"
 ✔ Check menu link with data set "Procedures -> Configuration menu link"
 ✔ Check menu link with data set "Procedures -> Load Compendium menu link"
 ✔ Check menu link with data set "Procedures -> Batch Results menu link"
 ✔ Check menu link with data set "Procedures -> Electronic Reports menu link"
 ✔ Check menu link with data set "Procedures -> Lab Documents menu link"
 ✔ Check menu link with data set "Admin -> Config menu link"
 ✔ Check menu link with data set "Admin -> Clinic -> Facilities menu link"
 ✔ Check menu link with data set "Admin -> Clinic -> Calendar menu link"
 ✔ Check menu link with data set "Admin -> Clinic -> Import Holidays menu link"
 ✔ Check menu link with data set "Admin -> Patients -> Patient Reminders menu link"
 ✔ Check menu link with data set "Admin -> Patients -> Merge Patients menu link"
 ✔ Check menu link with data set "Admin -> Patients -> Manage Duplicates menu link"
 ✔ Check menu link with data set "Admin -> Practice -> Practice Settings menu link"
 ✔ Check menu link with data set "Admin -> Practice -> Rules menu link"
 ✔ Check menu link with data set "Admin -> Practice -> Alerts menu link"
 ✔ Check menu link with data set "Admin -> Coding -> Codes menu link"
 ✔ Check menu link with data set "Admin -> Coding -> Native Data Loads menu link"
 ✔ Check menu link with data set "Admin -> Coding -> External Data Loads menu link"
 ✔ Check menu link with data set "Admin -> Forms -> Forms Administration menu link"
 ✔ Check menu link with data set "Admin -> Forms -> Layouts menu link"
 ✔ Check menu link with data set "Admin -> Forms -> Lists menu link"
 ✔ Check menu link with data set "Admin -> Documents -> Document Templates menu link"
 ✔ Check menu link with data set "Admin -> System -> Backup menu link"
 ✔ Check menu link with data set "Admin -> System -> Files menu link"
 ✔ Check menu link with data set "Admin -> System -> Language menu link"
 ✔ Check menu link with data set "Admin -> System -> Certificates menu link"
 ✔ Check menu link with data set "Admin -> System -> Logs menu link"
 ✔ Check menu link with data set "Admin -> System -> Audit Log Tamper menu link"
 ✔ Check menu link with data set "Admin -> System -> Diagnostics menu link"
 ✔ Check menu link with data set "Admin -> System -> API Clients menu link"
 ✔ Check menu link with data set "Admin -> Users menu link"
 ✔ Check menu link with data set "Admin -> Address Book menu link"
 ✔ Check menu link with data set "Admin -> ACL menu link"
 ✔ Check menu link with data set "Reports -> Clients -> List menu link"
 ✔ Check menu link with data set "Reports -> Clients -> Rx menu link"
 ✔ Check menu link with data set "Reports -> Clients -> Patient List Creation menu link"
 ✔ Check menu link with data set "Reports -> Clients -> Message List menu link"
 ✔ Check menu link with data set "Reports -> Clients -> Clinical menu link"
 ✔ Check menu link with data set "Reports -> Clients -> Referrals menu link"
 ✔ Check menu link with data set "Reports -> Clients -> Immunization Registry menu link"
 ✔ Check menu link with data set "Reports -> Clinic -> Report Results menu link"
 ✔ Check menu link with data set "Reports -> Clinic -> Standard Measures menu link"
 ✔ Check menu link with data set "Reports -> Clinic -> Automated Measures (AMC) menu link"
 ✔ Check menu link with data set "Reports -> Clinic -> 2024 Real World Testing Report menu link"
 ✔ Check menu link with data set "Reports -> Clinic -> 2023 Real World Testing Report menu link"
 ✔ Check menu link with data set "Reports -> Clinic -> Alerts Log menu link"
 ✔ Check menu link with data set "Reports -> Visits -> Daily Report menu link"
 ✔ Check menu link with data set "Reports -> Visits -> Patient Flow Board menu link"
 ✔ Check menu link with data set "Reports -> Visits -> Encounters menu link"
 ✔ Check menu link with data set "Reports -> Visits -> Appt-Enc menu link"
 ✔ Check menu link with data set "Reports -> Visits -> Superbill menu link"
 ✔ Check menu link with data set "Reports -> Visits -> Eligibility menu link"
 ✔ Check menu link with data set "Reports -> Visits -> Eligibility Response menu link"
 ✔ Check menu link with data set "Reports -> Visits -> Chart Activity menu link"
 ✔ Check menu link with data set "Reports -> Visits -> Charts Out menu link"
 ✔ Check menu link with data set "Reports -> Visits -> Services menu link"
 ✔ Check menu link with data set "Reports -> Visits -> Syndromic Surveillance menu link"
 ✔ Check menu link with data set "Reports -> Financial -> Sales menu link"
 ✔ Check menu link with data set "Reports -> Financial -> Cash Rec menu link"
 ✔ Check menu link with data set "Reports -> Financial -> Front Rec menu link"
 ✔ Check menu link with data set "Reports -> Financial -> Pmt Method menu link"
 ✔ Check menu link with data set "Reports -> Financial -> Collections and Aging menu link"
 ✔ Check menu link with data set "Reports -> Financial -> Pat Ledger"
 ✔ Check menu link with data set "Reports -> Financial -> Financial Summary by Service Code menu link"
 ✔ Check menu link with data set "Reports -> Financial -> Payment Processing menu link"
 ✔ Check menu link with data set "Reports -> Procedures -> Pending Res menu link"
 ✔ Check menu link with data set "Reports -> Procedures -> Statistics menu link"
 ✔ Check menu link with data set "Reports -> Insurance -> Distribution menu link"
 ✔ Check menu link with data set "Reports -> Insurance -> Indigents menu link"
 ✔ Check menu link with data set "Reports -> Insurance -> Unique SP menu link"
 ✔ Check menu link with data set "Reports -> Services -> Background Services menu link"
 ✔ Check menu link with data set "Reports -> Services -> Direct Message Log menu link"
 ✔ Check menu link with data set "Reports -> Services -> IP Tracker menu link"
 ✔ Check menu link with data set "Miscellaneous -> Dicom Viewer menu link"
 ✔ Check menu link with data set "Miscellaneous -> Patient Education menu link"
 ✔ Check menu link with data set "Miscellaneous -> Authorizations menu link"
 ✔ Check menu link with data set "Miscellaneous -> Chart Tracker menu link"
 ✔ Check menu link with data set "Miscellaneous -> Office Notes menu link"
 ✔ Check menu link with data set "Miscellaneous -> Batch Communication Tool menu link"
 ✔ Check menu link with data set "Miscellaneous -> New Documents menu link"

Check User Menu Links (OpenEMR\Tests\E2e\CheckUserMenuLinks)
 ✔ Check user link with data set "Settings user menu link"
 ✔ Check user link with data set "Change Password user menu link"
 ✔ Check user link with data set "MFA Management user menu link"
 ✔ Check user link with data set "About OpenEMR user menu link"
 ✔ Check user link with data set "Logout user menu link"

Create Staff (OpenEMR\Tests\E2e\CreateStaff)
 ✔ Check add user

Login (OpenEMR\Tests\E2e\Login)
 ✔ Check openEmr login page
 ✔ Url without token should redirect to login page
 ✔ Visitor with valid credential can be authenticated
 ✔ Visitor without valid credential is not authenticated

OK (103 tests, 109 assertions)

```

**Comment by bradymiller** [comment]
Testing well and bringing in

**Comment by stephenwaite** [comment]
smaller_excited_elephant

**Pull request merged**